### PR TITLE
Create Dockerfile.konflux

### PR DESCRIPTION
**Pull request by MohammadiIram**
Fixes #https://issues.redhat.com/browse/RHOAIENG-14192

### CODE REVIEW

**Comment by spolti** [comment]
is this ready for review?

**Comment by MohammadiIram** [comment]
> is this ready for review?

yes, please review the PR.

File: Dockerfile.konflux [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ARG SOURCE_CODE=.
+
+FROM registry.redhat.io/ubi8/go-toolset@sha256:4ec05fd5b355106cc0d990021a05b71bbfb9231e4f5bdc0c5316515edf6a1c96 AS stage
+
+# Define a build argument for the PNC list of built files
+ARG PNC_FILES_JSON
+RUN echo "Files to download: $PNC_FILES_JSON"
+
+# Install packages for the install script and extract archives
+RUN microdnf --setopt=install_weak_deps=0 --setopt=tsflags=nodocs install -y unzip jq wget
```
Comment (spolti) [inline Dockerfile.konflux:12]: don't we need the rpm.in and lock files to be able to install it?

File: Dockerfile.konflux [hunk 2]
```diff
@@ -0,0 +1,72 @@
+
+# Build arguments
+ARG SOURCE_CODE=.
+
+FROM registry.redhat.io/ubi8/go-toolset@sha256:4ec05fd5b355106cc0d990021a05b71bbfb9231e4f5bdc0c5316515edf6a1c96 AS stage
```
Comment (spolti) [inline Dockerfile.konflux:5]: can you please explain what this stage is used for?

File: Dockerfile.konflux [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+WORKDIR $STAGE_DIR
+
+# Filter the zip files only and unzip them in /root/
+RUN echo "$PNC_FILES_JSON" | jq -r '.[] | select(test("\\.zip$"))' | \
+    while read url; do wget --no-check-certificate "$url"; done && \
+    for file in *.zip; do unzip -d /root/ "$file"; done
+
+
+###############################################################################
+FROM registry.redhat.io/ubi8/openjdk-17-runtime@sha256:e2f33a6c60db4f4e70882a4a557eec5890997f8a1be7e3eb8971a0ff8a45a1a8 as runtime
```
Comment (spolti) [inline Dockerfile.konflux:24]: comment with the tag used

File: Dockerfile.konflux (hunk 1 above)
Comment (spolti) [reply Dockerfile.konflux:12]: is it to download files from PNC?

what about the konflux cache?

File: Dockerfile.konflux (hunk 2 above)
Comment (spolti) [reply Dockerfile.konflux:5]: why the base image h ere is go-toolset?

File: Dockerfile.konflux (hunk 1 above)
Comment (spolti) [reply Dockerfile.konflux:12]: where the java bits is being built?

File: Dockerfile.konflux [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      io.k8s.display-name="odh-modelmesh" \
+      io.k8s.description="odh-modelmesh" \
+      com.redhat.license_terms="https://www.redhat.com/licenses/Red_Hat_Standard_EULA_20191108.pdf"
+
+USER root
+
+RUN sed -i 's:security.provider.12=SunPKCS11:#security.provider.12=SunPKCS11:g' /usr/lib/jvm/java-17-openjdk-*/conf/security/java.security \
+    && sed -i 's:#security.provider.1=SunPKCS11 ${java.home}/lib/security/nss.cfg:security.provider.12=SunPKCS11 ${java.home}/lib/security/nss.cfg:g' /usr/lib/jvm/java-17-openjdk-*/conf/security/java.security
+
+ENV JAVA_HOME=/usr/lib/jvm/jre-17-openjdk
```
Comment (spolti) [inline Dockerfile.konflux:44]: the base image already provides this env, IMHO, we don't need to set it here.

File: Dockerfile.konflux [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+
+COPY --from=stage root/target/dockerhome/ /opt/kserve/mmesh/
+COPY --from=stage root/target/dockerhome/version /etc/modelmesh-version
+
+
+# Make this the current directory when starting the container
+WORKDIR /opt/kserve/mmesh
+
+RUN microdnf install shadow-utils
```
Comment (spolti) [inline Dockerfile.konflux:54]: don't we need to provide the rpm files?

File: Dockerfile.konflux [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+RUN useradd -c "Application User" -U -u ${USERID} -m app && \
+    chown -R app:0 /home/app && \
+    chmod g+w /etc/passwd && \
+    ln -s /opt/kserve/mmesh /opt/kserve/tas && \
+    mkdir -p log && \
+    chown -R app:0 . && \
+    chmod -R 771 . && chmod 775 *.sh *.py && \
+    echo "${CI_CONTAINER_VERSION}" > /opt/kserve/mmesh/build-version && \
+    sed -i 's/security.useSystemPropertiesFile=true/security.useSystemPropertiesFile=false/g' $JAVA_HOME/conf/security/java.security
```
Comment (spolti) [inline Dockerfile.konflux:63]: this line is to disable FIPS, do we want that?

File: Dockerfile.konflux (hunk 2 above)
Comment (MohammadiIram) [reply Dockerfile.konflux:5]: Yes, it should be ubi8/ubi-minimal.Updated it

File: Dockerfile.konflux (hunk 3 above)
Comment (MohammadiIram) [reply Dockerfile.konflux:24]: done

File: Dockerfile.konflux (hunk 5 above)
Comment (MohammadiIram) [reply Dockerfile.konflux:54]: yes, added it

File: Dockerfile.konflux (hunk 4 above)
Comment (MohammadiIram) [reply Dockerfile.konflux:45]: ok

File: Dockerfile.konflux (hunk 1 above)
Comment (spolti) [reply Dockerfile.konflux:12]: can you please clarify the other questions?

File: Dockerfile.konflux [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+WORKDIR $STAGE_DIR
+
+# Filter the zip files only and unzip them in /root/
+RUN echo "$PNC_FILES_JSON" | jq -r '.[] | select(test("\\.zip$"))' | \
+    while read url; do wget --no-check-certificate "$url"; done && \
+    for file in *.zip; do unzip -d /root/ "$file"; done
+
+
+###############################################################################
+#latest tag
```
Comment (spolti) [inline Dockerfile.konflux:24]: which is the latest tag?

please adopt a pattern. e.g.:

```
openjdk-17-runtime:1.17-9
```

**Pull request merged**